core/src/main/java/org/infinispan/globalstate/impl/EmbeddedLocalConfigurationManager.java

```diff
@@ -0,0 +1,167 @@
+import org.infinispan.commons.CacheConfigurationException;
+import org.infinispan.commons.api.CacheContainerAdmin;
+import org.infinispan.commons.util.concurrent.ConcurrentHashSet;
+import org.infinispan.configuration.ConfigurationManager;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.infinispan.configuration.global.GlobalStateConfiguration;
+import org.infinispan.configuration.parsing.ConfigurationBuilderHolder;
+import org.infinispan.configuration.parsing.ParserRegistry;
+import org.infinispan.eviction.PassivationManager;
+import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.factories.GlobalComponentRegistry;
+import org.infinispan.globalstate.GlobalConfigurationManager;
+import org.infinispan.globalstate.LocalConfigurationManager;
+import org.infinispan.jmx.CacheJmxRegistration;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.persistence.manager.PersistenceManager;
+import org.infinispan.util.DependencyGraph;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+/**
+ * The default implementation of {@link LocalConfigurationManager} for embedded scenarios.
+ *
+ * This component persists cache configurations to the {@link GlobalStateConfiguration#persistentLocation()} in a
+ * <pre>caches.xml</pre> file which is read on startup.
+ *
+ * @author Tristan Tarrant
+ * @since 9.2
+ */
+
+public class EmbeddedLocalConfigurationManager implements LocalConfigurationManager {
+   private static Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass());
+   private ConcurrentHashSet<String> persistentCaches = new ConcurrentHashSet<>();
+   private EmbeddedCacheManager cacheManager;
+   private ParserRegistry parserRegistry;
+   private GlobalConfiguration globalConfiguration;
+   private GlobalConfigurationManager globalConfigurationManager;
+
+   public EmbeddedLocalConfigurationManager() {
+   }
+
+   public void initialize(EmbeddedCacheManager cacheManager, GlobalConfigurationManager globalConfigurationManager) {
+      this.globalConfiguration = cacheManager.getCacheManagerConfiguration();
+      this.cacheManager = cacheManager;
+      this.globalConfigurationManager = globalConfigurationManager;
+      this.parserRegistry = new ParserRegistry();
+   }
+
+   @Override
+   public void validateFlags(EnumSet<CacheContainerAdmin.AdminFlag> flags) {
+      if (flags.contains(CacheContainerAdmin.AdminFlag.PERSISTENT) && !globalConfiguration.globalState().enabled())
+         throw log.globalStateDisabled();
+   }
+
+   public void createCache(String name, Configuration configuration, EnumSet<CacheContainerAdmin.AdminFlag> flags) {
+      Configuration existing = cacheManager.getCacheConfiguration(name);
+      if (existing == null) {
+         cacheManager.defineConfiguration(name, configuration);
```

Comment:
should check the returned configuration is the same as `configuration` (or null)

Reply:
Not quite sure I understand what you mean here. I already handle the case where the "new" config and the existing ones are equals, in the the second else clause by printing a debug log.

Reply:
forget :)